client/modules/workplace/components/collection_fields/object_tree.jsx

```diff
@@ -5,12 +5,51 @@ import TreeView from 'react-treeview';
 import ObjectItem from './object_item.jsx';
 import JoiningFields from '../../containers/joining_collections';
 
+const parseObj = obj => {
+  const parsedObj = obj;
+  if (parsedObj.nestedData) {
+    parsedObj.isCollapsed = true;
+    parsedObj.nestedData = parsedObj.nestedData.map(parseObj);
+    return parsedObj;
+  }
+  return obj;
+};
+
+const findFirstIndex = (data, expression) => {
+  return data.findIndex(innerArray =>
+    innerArray.some(collection =>
+      collection.name === expression.split('.')[0]
+    )
+  );
+};
+
+const toggleCollapse = (data, expression) => {
+  const firstIndex = findFirstIndex(data, expression);
+  const path = expression.split('.');
+  let link = data[firstIndex][0];
+
+  path.shift();
+  if (path.length) {
+    path.forEach(name => {
+      link = link.nestedData.find(nestedElement =>
+        nestedElement.name === name.replace('`', '').replace('`', ''));
+    });
+  }
+
+  if (link.hasOwnProperty('isCollapsed')) {
+    link.isCollapsed = !link.isCollapsed;
+  }
```

Comment:
@uapasha 
we can click on any element, like number, string, etc. And we should not create there field `isCollapsed`, cause it is only for arrays and objects.